### ISSUE TITLE
SIMclr training vs test sets configuration

### ISSUE THREAD

**Issue by Bontempogianpaolo1** [binli123/dsmil-wsi#39]
Hi @binli123 ,

I'm trying to replicate your results without success on camelyon16. I put the number of classes to 1 and also tried weights online for computing the feats on both training and test set. Even with that I still obtain only 0.7% AUC... So I start thinking about how I organized the data different from you. I downloaded the data from here: https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100439/CAMELYON16/
the data is divided into training and test. I used as threeshold 25 for filtering out background. So I used only the training set for training the self-supervised model.
After that, even with the model you published on drive,  I extracted feats with the compute_feat script for both training and test(especially with the fusion option). Finally, I modified the train_tcga for considering them as sources for the training set and the test set (270 /130 bags).  Even 

If instead, I use the features precomputed by you the mil model works. So the problem could be how I split data or how I extract embeddings. What am I missing?

**Comment by binli123** [comment]
Could you check out the CSV files containing the features and labels？

**Comment by Bontempogianpaolo1** [comment]
the csv seems correct...  Here some screenshots of embeddings extracting using your pretrained model found at https://drive.google.com/drive/folders/1_mumfTU3GJRtjfcJK_M0fWm048sYYFqi on patches extracted using 19 as threeshold:

camelyon.csv
<img width="215" alt="image" src="https://user-images.githubusercontent.com/28188499/169819047-f6adb5cc-48c6-4df2-86dd-626eacbd790d.png">
normal143.csv
<img width="899" alt="image" src="https://user-images.githubusercontent.com/28188499/169819785-03ba6de2-5f2d-4cdb-a972-5629605242d1.png">


However comparing your features with mine the number of rows is different...So is it possible that the number of patches is influencing the results? Here the number of patches using different background thresholds for 5 different slides:

| Slide name | th =19 | th=25| your features|
| ------------- | ------------- |------------- |------------- |
|tumor_108 | 29905 | 402 | 23263|
| test_124  | 6693 | 3001 | 2402 |
|tumor_095 | 39960 | 1002 | 31791|
|normal_137| 33396 | 505 | 23443|
|tumor_076 | 61670 | 42057 | 19708|


Maybe is the image quality not correct for your embedder? Here an example of patch extracted at level=0 magnitude=20
<img width="120" alt="image" src="https://user-images.githubusercontent.com/28188499/169820939-fce08e50-3508-4769-9dca-7776326eb106.png">
<img width="123" alt="image" src="https://user-images.githubusercontent.com/28188499/169821091-d7357fe7-2b76-44ca-b4ed-ad29c8f3ba23.png">

Thanks in advance for your reply

**Comment by Bontempogianpaolo1** [comment]
I took directly our embedder without training and I passed it to the compute_feats script with InstanceNorm2d since it is the default parameter

**Comment by binli123** [comment]
> model_v2.pth

Have you tried model_v0.pth and model_v1.pth, did they also not work?

**Comment by Bontempogianpaolo1** [comment]
not yet... I considered the v2 model as the best one

**Comment by Bontempogianpaolo1** [comment]
screenshot features using model-v0
<img width="948" alt="image" src="https://user-images.githubusercontent.com/28188499/169862452-9002230c-8a7b-4774-aedc-88c8aaf34cf9.png">

screenshot features using model-v1

<img width="941" alt="image" src="https://user-images.githubusercontent.com/28188499/169863692-93e03f47-5091-44f2-b5de-a6635ed1b595.png">

**Comment by binli123** [comment]
![normal_141_42_54](https://user-images.githubusercontent.com/25302567/169871070-6c510eea-07ca-4a8c-87b6-2d95fa58875c.jpg)
Does your `normal_141_42_54.jpg` look like this?
My feature csv using v2
[normal_141.csv](https://github.com/binli123/dsmil-wsi/files/8756791/normal_141.csv)

**Comment by Bontempogianpaolo1** [comment]
I don't have it... What are the parameters you used for the script deepzoom_tiler.py  in the case of Camelyon?

This is my normal_141 48_112.jpeg
<img width="118" alt="image" src="https://user-images.githubusercontent.com/28188499/169873148-2e1311f5-9ef1-4a5a-ab41-90625f1d446e.png">

**Comment by binli123** [comment]
It turns out that Camleyon16 consists of mixed magnifications, so by experimenting the correct configuration: `python deepzoom_tiler.py -m 1 -b 20 -d Camelyon16-pilot -v tif`

**Comment by binli123** [comment]
> In this way the magnitude become x10 right? is your embedder trained under this magnitude? Since it is inside the folder called x20 I didn't expect it

I think it is still 20x because the base magnification has ~0.25 micro/pixel which corresponds to 40x for the Aperio scanner (FDA standard). A 20x magnification corresponds to ~0.5 micron/pixel. Camelyon16 uses a mixture of magnifications with different micron/pixel. 

![image](https://user-images.githubusercontent.com/25302567/169899028-be55756b-b9fc-4ac7-9d52-37173feb3ded.png)

Notice how their 20x and 40x scanners have almost the same micron/pixel? You will call the "20x" RUMC a "40x" image for UMCU. So better just use the FDA standard.

**Comment by Bontempogianpaolo1** [comment]
Ok! I'm just trying it and inside the folder "temp" the patches are stored inside a "10" folder ( imagining referring to the magnitude). Anyway thank you very much for your replies! I'll just try the entire pipeline again with these new patches and I'll tell you the results as soon as possible

**Comment by Bontempogianpaolo1** [comment]
It worked !! But I still have problems :(... I'm opening a new issue for that since it is not relative to the dataset but to the embedder
